arch/arm64/boot/dts/qcom/sdm450-xiaomi-rosy.dts

```diff
@@ -31,6 +31,47 @@
 			no-map;
 		};
 	};
+
+	i2c-sensors {
+		compatible = "i2c-gpio";
+		sda-gpios = <&tlmm 14 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&tlmm 15 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		i2c-gpio,delay-us = <2>; /* ~100 kHz */
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ak09918: magnetometer@c {
+			compatible = "asahi-kasei,ak09918";
+			reg = <0x0c>;
+
+			vdd-supply = <&pm8953_l10>;
+			vid-supply = <&pm8953_l6>;
+		};
+
+		/*
+		 * Driver still missing
+		 * Just for general information
+		 */
+		ltr579: light@53 {
```

Comment:
i am not sure it hase ltr579 variant, my rosy uses stk3x1x. There is an st3310 driver in the kernel.

Reply:
I can try with st3310; I have no idea which one my rosy uses

Reply:
You can check your device with smgr